File: openbis_api/source/java/ch/ethz/sis/openbis/generic/asapi/v3/dto/externaldms/ExternalDmsType.java
package ch.ethz.sis.openbis.generic.asapi.v3.dto.externaldms;

import com.fasterxml.jackson.annotation.JsonIgnore;

import ch.systemsx.cisd.base.annotation.JsonObject;

@JsonObject("as.dto.externaldms.ExternalDmsType")
public enum ExternalDmsType
{
    OPENBIS, GIT, UNDEFINED;

    @JsonIgnore
    public static ExternalDmsType fromString(String str)
    {
        switch (str)
        {
            case "OPENBIS":
                return ExternalDmsType.OPENBIS;
            case "GIT":
                return ExternalDmsType.GIT;
            default:
                return ExternalDmsType.UNDEFINED;
        }
    }
}

File: openbis/source/java/ch/systemsx/cisd/openbis/public/resources/api/v3/as/dto/externaldms/ExternalDmsType.js
/**
 * @author anttil
 */
 
define([ "stjs", "as/dto/common/Enum" ], function(stjs, Enum) {
	var ExternalDmsType = function() {
		Enum.call(this, [ "OPENBIS", "GIT", "UNDEFINED" ]);
	};
	stjs.extend(ExternalDmsType, Enum, [ Enum ], function(constructor, prototype) {
	}, {});
	return new ExternalDmsType();
})